Content.Shared/_Floof/ModifyUndies/ModifyUndiesSystem.cs

```diff
@@ -0,0 +1,188 @@
+using System.Linq;
+using Content.Shared.DoAfter;
+using Content.Shared.Humanoid;
+using Content.Shared.Humanoid.Markings;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Inventory;
+using Content.Shared.Popups;
+using Content.Shared.Verbs;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.FloofStation.ModifyUndies;
+
+/// <summary>
+/// System that controls removing and adding underwear to humanoid entities through a verb.
+/// </summary>
+public sealed class ModifyUndiesSystem : EntitySystem
+{
+    [Dependency] private readonly MarkingManager _markingManager = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedHumanoidAppearanceSystem _humanoid = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly InventorySystem _inventory = default!;
+    [Dependency] private readonly IEntityManager _entMan = default!;
+    [Dependency] private readonly INetManager _net = default!;
+
+    public static readonly VerbCategory UndiesCat = new("verb-categories-undies", "/Textures/Interface/VerbIcons/undies.png");
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ModifyUndiesComponent, GetVerbsEvent<Verb>>(AddModifyUndiesVerb);
+        SubscribeLocalEvent<ModifyUndiesComponent, ModifyUndiesDoAfterEvent>(ToggleUndies);
+    }
+
+    private void AddModifyUndiesVerb(Entity<ModifyUndiesComponent> ent, ref GetVerbsEvent<Verb> args)
+    {
+        if (args.Hands == null || !args.CanAccess || !args.CanInteract)
+            return;
+
+        if (!TryComp<HumanoidAppearanceComponent>(args.Target, out var humApp))
+            return;
+
+        if (args.User != args.Target && _inventory.TryGetSlotEntity(args.Target, "jumpsuit", out _))
+            return; // mainly so people cant just spy on others undies *too* easily
```

Comment:
I'd recommend to use the slot blocker here because it's way more versatile. It will also come with the added benefit of preventing a player from toggling someone's underwear if they are e.g. wearing just an outer clothing piece (unless it's tagged with Skirt iirc)

Note that this will require adding some `using` statements, so please don't just blindly use "commit suggestion" here.
```suggestion
        if (!TryComp<HumanoidAppearanceComponent>(args.Target, out var humApp) || !TryComp<InventoryComponent>(args.Target, out var inventory))
            return;

        if (args.User != args.Target && if (_slotBlocker.IsSlotObstructed((args.Target, inventory), null, SlotBlockerSystem.CheckType.IgnoreBlockerPreference, SlotFlags.INNERCLOTHING, out _))
            return; // mainly so people cant just spy on others undies *too* easily
```